handout/testing/Introduction to TDD Testing.md

```diff
@@ -0,0 +1,185 @@
+
+```
+...
+"scripts": {
+	"test": "karma start",
+	...
+}
+...
+```
+Now running `npm test` will start Karma. Below is the output of our Karma test, as you can see we had one test that passed, running in a Chrome 48 browser. 
+
+![image](images/simple-output.png)
+
+## Simple Test
+
+To begin, lets start by writing a simple test in Jasmine.
+
+```typescript 
+import {
+  describe,
+  expect,
+  it
+} from 'angular2/testing';
+
+describe('Testing math', () => {
+
+  it('multiplying should work', () => {
+  
+  		expect(4*4).toEqual(16));
+  });
+});
+```
+
+Though this test may be trivial, it illustrates the basic elements of a unit test. We explain what this test is for by using `describe`, and we use `it` to assert what kind of result we are expecting from our test. These are user defined so its a good idea to be as descriptive and as accurate in these messages as possible. Messages like "should work", or "testing service" don't really explain exactly whats going on and may be confusing when running multiple tests. 
+
+Our actual test is basic, we use `expect` to formulate a scenario and use `toEqual` to assert the resulting condition we are expecting from such scenario. The test will pass if our assertion is equal to the resulting condition, and fail otherwise. You always want your tests to pass - do not write tests that have the results you want in a failed state. 
+
+### Using Chia 
+Chia is an assertion library with some tasty syntax sugar that can be paired with any other testing framework. It lets us write tests in a TDD (Test Driven Development) style or BDD (Behaviour Driven Development) style. We already know what TDD is (read the intro!), so what exactly is BDD? Well BDD is the combination of using TDD with natural language constructs (english like sentences) to express the behaviour and outcomes of unit tests. Jasmine already uses a TDD style, so we'll be using Chia for its BDD interfaces, mainly through the use of `should`, and `expect`. 
+
+```typescript
+import {
+	describe,
+	expect,
+	it
+} from 'angular2/testing';
+
+describe('Testing math', () => {
+
+	it('multiplying should work', () => {
+		
+        let testMe = 16;
+
+		// Using the expect interface
+        chai.expect(testMe).to.be.a('number');
+        chai.expect(testMe).to.equal(16);
+
+		// Using the should interface
+        chai.should();
```

Comment:
It sets up the `should` interface, I got it from the code examples on the homepage: [http://chaijs.com/](http://chaijs.com/)